src/app/progress/quest.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import * as _ from 'underscore';
+import { IDestinyObjectiveProgress, IDestinyInventoryComponent, IDestinyItemInstanceComponent, IDestinyItemComponent } from '../bungie-api/interfaces';
+import { sum } from '../util';
+import { t } from 'i18next';
+import { percent } from '../inventory/dimPercentWidth.directive';
+import './quest.scss';
+
+interface QuestProps {
+  defs;
+  item: IDestinyItemComponent;
+  objectives: IDestinyObjectiveProgress[];
+}
+
+export function Quest(props: QuestProps) {
+  const { defs, item, objectives } = props;
+
+  const itemDef = defs.InventoryItem.get(item.itemHash);
+
+  const percentComplete = sum(objectives, (objective) => {
+    const objectiveDef = defs.Objective.get(objective.objectiveHash);
+    if (objectiveDef.completionValue) {
+      return Math.min(1.0, (objective.progress || 0) / objectiveDef.completionValue) / objectives.length;
+    } else {
+      return 0;
+    }
+  });
+
+  return <div className="milestone-quest item-quest">
+    <div className="milestone-icon">
+      <img src={`https://www.bungie.net${itemDef.displayProperties.icon}`} />
```

Comment:
Could a `<BungieImage />` component could be useful for every time you do this?

```
export default function BungieImage({ src }) {
  return <img src={`https://www.bungie.net${src}`} />
}
```

Reply:
Yeah, that'll be super helpful. We have similar helpers in Angular, but I was just starting with the most direct thing that'd work before looking for patterns I can refactor out.